### PR TITLE
CLOUDP-58851: Include readiness probe in Agent image

### PR DESCRIPTION
**Pull request by chatton**
### All Submissions:

* [ ] Have you opened an Issue before filing this PR?
* [X] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [X] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?
* [ ] Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if such).

This PR updates the agent image to download the readiness probe binary and configure the readiness probe in the agent stateful set.

### CODE REVIEW

File: cmd/testrunner/crds/crds.go [hunk 1]
```diff
@@ -64,7 +64,7 @@ func allCrds(deployDir string) ([]string, error) {
 	crdDir := path.Join(deployDir, "crds")
 	var crdFilePaths []string
 	err := filepath.Walk(crdDir, func(path string, info os.FileInfo, err error) error {
-		if strings.HasSuffix(info.Name(), "_crd.yaml") {
+		if info != nil && strings.HasSuffix(info.Name(), "_crd.yaml") {
```
Comment (chatton) [inline cmd/testrunner/crds/crds.go]: info can be nil if there are other directories in `crdDir`

File: pkg/controller/mongodb/mongodb_controller.go [hunk 2]
```diff
@@ -210,16 +213,21 @@ func buildAutomationConfigConfigMap(mdb mdbv1.MongoDB) (corev1.ConfigMap, error)
 func buildContainers(mdb mdbv1.MongoDB) ([]corev1.Container, error) {
 	agentCommand := []string{
 		"agent/mongodb-agent",
-		"-cluster=/var/lib/automation/config/automation-config",
+		"-cluster=" + clusterFilePath,
 		"-skipMongoStart",
 		"-noDaemonize",
+		"-healthCheckFilePath=" + agentHealthStatusFilePath,
+		"-serveStatusPort=5000",
```
Comment (chatton) [inline pkg/controller/mongodb/mongodb_controller.go]: the agent won't write the health status file correctly if this is not configured

File: test/e2e/mongodbtests/mongodbtests.go [hunk 3]
```diff
@@ -0,0 +1,101 @@
+package mongodbtests
```
Comment (chatton) [inline test/e2e/mongodbtests/mongodbtests.go]: this package contains functions that return test functions which will be called by the framework for commonly tested things

**Comment by chatton** [comment]
@rodrigovalin 

I've added the test which kills a pod and ensures the StatefulSet gets back into ready state and the replica set is connectable. I think it makes the most sense to add the other test you mentioned in a future PR as this one has grown quite a bit due to the changes to the test runner/evergreen.yaml to run different individual tests

**Comment by rodrigovalin** [comment]
@chatton I agree that this PR should not recieve more changes now.

**Pull request merged**